.changelog/19218.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+resource: lowercase names enforced for v2 resources only.
```

Comment:
Is this considered a breaking change? Also could we incorporate this language in the release note? 

With this change we will only accept lowercase alphanumeric characters and - , and names that start and end with an alphanumeric character. All other characters will be considered incompatible.